lib/rom/sql/extensions/postgres/types/ltree.rb

```diff
@@ -0,0 +1,140 @@
+module ROM
+  module SQL
+    module Postgres
+      module Values
+        LabelPath = ::Struct.new(:path) do
+          def labels
+            path.split('.')
+          end
+        end
+      end
+
+      # @api public
+      module Types
+        # @see https://www.postgresql.org/docs/current/static/ltree.html
+
+        LTree = Type('ltree') do
+          SQL::Types.define(Values::LabelPath) do
+            input do |label_path|
+              label_path ? label_path.path : nil
+            end
+
+            output do |label_path|
+              label_path ? Values::LabelPath.new(label_path.to_s) : nil
+            end
+          end
+        end
+
+        module LTreeMethods
+          ASCENDANT = ["(".freeze, " @> ".freeze, ")".freeze].freeze
+          FIND_ASCENDANT = ["(".freeze, " ?@> ".freeze, ")".freeze].freeze
+          DESCENDANT = ["(".freeze, " <@ ".freeze, ")".freeze].freeze
+          FIND_DESCENDANT = ["(".freeze, " ?<@ ".freeze, ")".freeze].freeze
+          MATCH_ANY = ["(".freeze, " ? ".freeze, ")".freeze].freeze
+          MATCH_ANY_LQUERY = ["(".freeze, " ?~ ".freeze, ")".freeze].freeze
+          MATCH_LTEXTQUERY = ["(".freeze, " @ ".freeze, ")".freeze].freeze
+          MATCH_ANY_LTEXTQUERY = ["(".freeze, " ?@ ".freeze, ")".freeze].freeze
+
+          def match(type, expr, query)
+            Attribute[SQL::Types::Bool].meta(sql_expr: Sequel::SQL::BooleanExpression.new(:'~', expr, query))
+          end
+
+          def match_any(type, expr, query)
+            array = build_array_query(query)
+            Attribute[SQL::Types::Bool].meta(sql_expr: custom_sql_expr(MATCH_ANY, expr, array))
+          end
+
+          private
+
+          def custom_sql_expr(string, expr, query)
```

Comment:
If this is used only for operators then having a more specific name like ‘operator_expr’ would be a better choice